x-pack/plugins/security_solution/public/timelines/components/timeline/helpers.tsx

```diff
@@ -253,3 +271,71 @@ export const focusUtilityBarAction = (containerElement: HTMLElement | null) => {
 export const resetKeyboardFocus = () => {
   document.querySelector<HTMLAnchorElement>('header.headerGlobalNav a.euiHeaderLogo')?.focus();
 };
+
+const buildISQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: string | number;
+}): string => {
+  if (isFieldTypeNested) {
+    return convertNestedFieldToQuery(field, value, browserFields);
+  } else if (checkIfFieldTypeIsDate(field, browserFields)) {
+    return convertDateFieldToQuery(field, value);
+  } else {
+    return `${field} : ${isNumber(value) ? value : escapeQueryValue(value)}`;
+  }
+};
+
+const buildEXISTSQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+}): string => {
+  return isFieldTypeNested
+    ? convertNestedFieldToExistQuery(field, browserFields)
+    : `${field} ${EXISTS_OPERATOR}`;
+};
+
+const buildISONEOFQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: Array<string | number>;
+}): string => {
+  if (isFieldTypeNested) {
+    // some kind of nested handler implementation if needed, I assume with BrowserFields?
+    return '';
+  } else {
+    // "['open', 'closed']"
+
+    return `${field} : ${JSON.stringify(value)
+      .replace(/\[/g, '(')
+      .replace(/\]/g, ')')
+      .replace(/,/g, ' OR ')}`;
+  }
+};
+
+function isStringOrNumberArray(
+  val: string | number | Array<string | number>
+): val is Array<string | number> {
+  return (
+    typeof val !== 'string' &&
```

Comment:
You could probably simplify this with `Array.isArray(val) && (typeof val[0] === 'string' || typeof val[0] === 'number')`

Reply:
changed